### PR TITLE
Make level-3

### PR DESCRIPTION
**Pull request by creator-max**
Add Engines. Update Vehicle, override Equals() for Vehicle. Update VehicleHelper.

### CODE REVIEW

File: AutoPark/VehicleHelper.cs [hunk 1]
```diff
@@ -34,5 +34,18 @@ public static void PrintVehicles(IReadOnlyList<Vehicle> vehicles)
                 Console.WriteLine(vehicle);
             }
         }
+
+        public static void PrintEqualsVehicles(IReadOnlyList<Vehicle> vehicles)
+        {
+            for (int i = 0; i < vehicles.Count - 1; i++)
+            {
+                for (int j = i + 1; j < vehicles.Count; j++)
```
Comment (kuzmiich) [inline AutoPark/VehicleHelper.cs:42]: Redundant second loop

File: AutoPark/ElectricalEngine.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿namespace AutoPark
+{
+    public class ElectricalEngine : Engine
+    {
+        public ElectricalEngine(double electricityConsumption)
+            : base("Electrical", 0.1d)
```
Comment (kuzmiich) [inline AutoPark/ElectricalEngine.cs:6]: Remove 0.1`d`, or add to DieselEngine

File: AutoPark/Vehicle.cs [hunk 3]
```diff
@@ -52,5 +55,16 @@ public int CompareTo(Vehicle other)
             else
                 throw new ArgumentNullException(nameof(other));
         }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+                return false;
+
+            if (!(obj is Vehicle other))
```
[the remark this answers is not in the document]
Comment (creator-max) [reply AutoPark/Vehicle.cs:64]: Thanks, fixed

File: AutoPark/ElectricalEngine.cs (hunk 2 above)
Comment (creator-max) [reply AutoPark/ElectricalEngine.cs:6]: Thanks, fixed. Added "d" everywhere

**Review by kuzmiich — APPROVED** [review]
Cool!

File: AutoPark/CombustionEngine.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    public class CombustionEngine : Engine
+    {
+        private const double CoefficientFuelConsumption = 100d;
+        public CombustionEngine(string typeName, double taxCoefficient)
+            : base(typeName, taxCoefficient)
+        {
+
+        }
+        public double EngineCompacityInCubCM { get; private protected set; }
+        public double FuelConsumptionPer100 { get; private protected set; }
```
Comment (PushMeNow) [inline AutoPark/CombustionEngine.cs:12]: Don't see the sense in private protected

File: AutoPark/Vehicle.cs [hunk 5]
```diff
@@ -52,5 +55,13 @@ public int CompareTo(Vehicle other)
             else
                 throw new ArgumentNullException(nameof(other));
         }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+                return false;
```
Comment (PushMeNow) [inline AutoPark/Vehicle.cs:62]: Redundant because  below you have <code>obj is Vehicle</code> condition

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: AutoPark/CombustionEngine.cs (hunk 4 above)
Comment (creator-max) [reply AutoPark/CombustionEngine.cs]: Thanks, fixed

File: AutoPark/Vehicle.cs (hunk 5 above)
Comment (creator-max) [reply AutoPark/Vehicle.cs]: Thanks, fixed

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**